src/main/java/uk/gov/hmcts/darts/audio/validation/MediaHideOrShowValidator.java

```diff
@@ -0,0 +1,60 @@
+package uk.gov.hmcts.darts.audio.validation;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.audio.exception.AudioApiError;
+import uk.gov.hmcts.darts.audio.model.MediaHideRequest;
+import uk.gov.hmcts.darts.common.component.validation.Validator;
+import uk.gov.hmcts.darts.common.entity.ObjectAdminActionEntity;
+import uk.gov.hmcts.darts.common.entity.ObjectHiddenReasonEntity;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+import uk.gov.hmcts.darts.common.repository.ObjectAdminActionRepository;
+import uk.gov.hmcts.darts.common.repository.ObjectHiddenReasonRepository;
+import uk.gov.hmcts.darts.common.validation.IdRequest;
+
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@RequiredArgsConstructor
+public class MediaHideOrShowValidator implements Validator<IdRequest<MediaHideRequest>> {
+
+    private final ObjectAdminActionRepository objectAdminActionRepository;
+    private final MediaIdValidator mediaIdValidator;
+    private final ObjectHiddenReasonRepository objectHiddenReasonRepository;
+
+    @Override
+    @SuppressWarnings("java:S5411")
+    public void validate(IdRequest<MediaHideRequest> request) {
+        mediaIdValidator.validate(request.getId());
+
+         if (request.getPayload().getIsHidden() && request.getPayload().getAdminAction() == null) {
```

Comment:
`request.getPayload()` and `request.getPayload().getAdminAction()` used a lot in the next few lines, consider making variables for them.

Reply:
you've onyl replaced one instance instead of all 3 of `request.getPayload().getAdminAction()`
![image](https://github.com/hmcts/darts-api/assets/131666106/7f9e5927-2a01-41cc-afd7-06e5045352d0)

and `request.getPayload().getIsHidden()` is used 3 times also:-
![image](https://github.com/hmcts/darts-api/assets/131666106/6bae81a0-acc7-4876-8cda-f04b9483eec5)